packages/node-core/src/blockchain.service.ts

```diff
@@ -58,6 +58,7 @@ export interface IBlockchainService<
   // Unfinalized blocks
   getHeaderForHash(hash: string): Promise<Header>;
   getHeaderForHeight(height: number): Promise<Header>;
+  getRequiredHeaderForHeight(height: number): Promise<Header & {timestamp: Date}>;
```

Comment:
I think this can be removed. Lets make timestamp required on Header. There's only one chain that we cant get the timestamp and its not worth making compromises just for that